src/components/CompletedSubReqCourse.vue

```diff
@@ -1 +1,184 @@
+          :uniqueID="courseObject.uniqueID"
+          :compact="true"
+          :active="false"
+          :isCompletedReqCourse="true"
+          :isReqCourse="true"
+          class="completed-reqCourses-course-object"
+        />
+        <div class="completed-reqCourses-course-object-semester">
+          {{ semesterLabel }}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import firebase from 'firebase/app';
+import Course from '@/components/Course.vue';
+import { DisplayableRequirementFulfillment, CourseTaken } from '@/requirements/types';
+import {
+  AppCourse,
+  FirestoreSemesterCourse,
+  AppSemester,
+  FirestoreSemesterType,
+} from '@/user-data';
+
+Vue.component('course', Course);
+
+type CrseInfo = {
+  roster: string;
+  crseIds: number[];
+};
+
+type CompletedSubReqCourseSlot = {
+  isCompleted: true;
+  courses: readonly CourseTaken[];
+};
+
+type IncompleteSubReqCourseSlot = {
+  isCompleted: false;
+  courses: CrseInfo[];
+};
+
+type SubReqCourseSlot = CompletedSubReqCourseSlot | IncompleteSubReqCourseSlot;
+
+type Data = {
+  courseObjects: AppCourse[];
+  semesterType: FirestoreSemesterType;
+  semesterYear: number;
+};
+
+export default Vue.extend({
+  props: {
+    subReq: Object as PropType<DisplayableRequirementFulfillment>,
+    subReqCourseId: Number,
+    crsesTaken: Array as PropType<readonly CourseTaken[]>,
+    semesters: Array as PropType<readonly AppSemester[]>,
+  },
+  data(): Data {
+    return {
+      semesterType: 'Fall',
+      semesterYear: 1234,
+      courseObjects: [],
+    };
+  },
+  mounted() {
+    const crseTaken = this.crsesTaken[0];
+
+    for (let i = 0; this.courseObjects.length < 1 && i < this.semesters.length; i += 1) {
+      const semester = this.semesters[i];
+      const filteredSemesterCourses = semester.courses.filter(
+        course =>
+          course.crseId === crseTaken.courseId &&
+          course.subject === crseTaken.subject &&
+          course.number === crseTaken.number
+      );
+      if (filteredSemesterCourses.length > 0) {
+        this.courseObjects.push(filteredSemesterCourses[0]);
+        this.semesterType = semester.type;
+        this.semesterYear = semester.year;
+      }
+    }
+  },
+  computed: {
+    courseLabel() {
+      return `Course ${this.subReqCourseId + 1}`;
+    },
+    resetText() {
+      return 'Reset';
+    },
+    semesterLabel() {
+      return `in ${this.$data.semesterType} ${this.$data.semesterYear}`;
+    },
+  },
+  methods: {
+    dragListener(event: { preventDefault: () => void }) {
+      if (!this.$data.scrollable) event.preventDefault();
+    },
+  },
+});
+</script>
+
+<style scoped lang="scss">
+@import '@/assets/scss/_variables.scss';
+
+.separator {
+  height: 1px;
+  width: 100%;
+  background-color: $inactiveGray;
+}
+
+.completed {
+  &-reqCourses {
+    &-course {
+      &-wrapper {
+        margin-top: 4%;
+        margin-bottom: 4%;
```

Comment:
I don't believe that percentage values for margins or paddings are great practice. Is there a strong reason they're used here instead of definate values instead? If not, can we switch them?